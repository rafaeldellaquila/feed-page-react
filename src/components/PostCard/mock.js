const postCardMock = {
  title: 'sunt aut facere repellat provident',
  body: 'quia et suscipit suscipit recusandae consequuntur',
  cover: 'https://via.placeholder.com/600/92c952',
  id: 0
}

export default postCardMock
