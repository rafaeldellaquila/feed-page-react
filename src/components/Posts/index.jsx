import P from 'prop-types'
import PostCard from '../PostCard'
import './styles.css'

const Posts = ({ posts = [] }) => {
  return (
    <div className="posts">
      {posts.map((post) => (
        <PostCard key={post.id} title={post.title} body={post.body} cover={post.cover} id={post.id} />
      ))}
    </div>
  )
}

Posts.propTypes = {
  posts: P.arrayOf(
    P.shape({
      title: P.string.isRequired,
      body: P.string.isRequired,
      cover: P.string.isRequired,
      id: P.number.isRequired
    })
  )
}
export default Posts
