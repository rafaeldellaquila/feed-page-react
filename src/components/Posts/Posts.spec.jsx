import { render, screen } from '@testing-library/react'
import Posts from '.'
import { postCardProps } from '../PostCard/mock'

const props = {
  posts: [
    {
      ...postCardProps,
    },
    {
      ...postCardProps,
      id: postCardProps.id + 1,
    },
    {
      ...postCardProps,
      id: postCardProps.id + 2,
    },
  ],
}

describe('<Post />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />)

    expect(screen.getAllByRole('heading', { name: /sunt/i })).toHaveLength(3)
    expect(screen.getAllByRole('img', { name: /sunt/i })).toHaveLength(3)
    expect(screen.getAllByText(/quia/i)).toHaveLength(3)
  })

  it('match the snapshot', () => {
    const { container } = render(<Posts {...props} />)
    expect(container).toMatchSnapshot()
  })
})
