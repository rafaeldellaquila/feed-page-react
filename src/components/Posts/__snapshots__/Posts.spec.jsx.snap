// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post /> match the snapshot 1`] = `
<div>
  <div
    class="posts"
  >
    <div
      class="post"
    >
      <img
        alt="sunt aut facere repellat provident"
        src="https://via.placeholder.com/600/92c952"
      />
      <div
        class="post-content"
      >
        <h2>
          sunt aut facere repellat provident
        </h2>
        <p>
          quia et suscipit suscipit recusandae consequuntur
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="sunt aut facere repellat provident"
        src="https://via.placeholder.com/600/92c952"
      />
      <div
        class="post-content"
      >
        <h2>
          sunt aut facere repellat provident
        </h2>
        <p>
          quia et suscipit suscipit recusandae consequuntur
        </p>
      </div>
    </div>
    <div
      class="post"
    >
      <img
        alt="sunt aut facere repellat provident"
        src="https://via.placeholder.com/600/92c952"
      />
      <div
        class="post-content"
      >
        <h2>
          sunt aut facere repellat provident
        </h2>
        <p>
          quia et suscipit suscipit recusandae consequuntur
        </p>
      </div>
    </div>
  </div>
</div>
`;
